### ISSUE TITLE
Entity moves between devices upon reboot

### ISSUE THREAD

**Issue by rb1111-1** [bzellman/WindmillAC#17]
I have 3 Windmill AC devices added to HA. However, only one entity is available. Two things I want to flag:

1. After providing the Auth Token, it takes a full reboot for HA to recognize the device. Not a problem, but mentioning as it may be related to the following:
2. After each reboot, the entity moves from device to device in seemingly random order. Whichever device the entity lands on, that is the device the entity controls (regardless of which Auth Token was provided to set up that entity's original device). 

Is it possible to have more than one device (and thus entities) with this integration?

HA 2024.7.3

Thank you!

**Comment by bzellman** [comment]
Thanks @rb1111-1 . 

I didn't build in multi-device use cases and was an oversight to not guard against as a consideration.

I'll pick this up and look at adding multi entity support. If for some reason there looks like a long runway on that. I'll put up a guard to better set use-case / expecations

**Comment by rb1111-1** [comment]
Thank you! When you get to it, I am happy to test for you.

**Comment by cbowns** [comment]
Same here, I have three devices I wanted to configure.

**Comment by jessel92** [comment]
Yeah, me as well. Multi-device support would be amazing since I have two windmills personally. But other than this issue, everything works great!

**Comment by rb1111-1** [comment]
Hi @bzellman — wanted to check in to see if you had a chance to look at implementing multiple devices. Spring is progressing and we're getting close to putting the window units in.

**Comment by Kthwaits** [comment]
Same here! Would love to be able to control both of my units with this integration now that we're entering the A/C time of year.

**Comment by bzellman** [comment]
Thank you @Kthwaits  for the PR. Merged to Master and Versions bumped.
